web/public/js/controllers/dojos-map-controller.js

```diff
@@ -103,10 +104,17 @@ function cdDojosMapCtrl($scope, $window, $state, $stateParams, $translate, $geol
     $state.go('dojo-detail',{country:country, path:path});
   }
 
+  $scope.getDojoURL = function(dojo) {
+    if (dojo) {
+      var urlSlug = dojo.url_slug || dojo.urlSlug;
+      return "/dojo/" + urlSlug;
+    }
+  }
+
   $scope.getDojo = function (marker) {
     var dojoId = marker.dojoId;
-    cdDojoService.load(dojoId, function (response) {
-      $scope.viewDojo(response);
+    return cdDojoService.load(dojoId, function (response) {
+      return $scope.getDojoURL(response);
```

Comment:
Unfortunately this won't work as the getDojo function will have already returned the response of cdDojoService.load before the Dojo details have even loaded.

I think we may actually need to leave the map links as they are for now, as we need to asynchronously get the Dojo URL at the time the marker is selected.